odoo/tools/sql.py

```diff
@@ -24,7 +24,7 @@ def existing_tables(cr, tablenames):
           JOIN pg_namespace n ON (n.oid = c.relnamespace)
          WHERE c.relname IN %s
            AND c.relkind IN ('r', 'v', 'm')
-           AND n.nspname = 'public'
+           AND n.nspname = (SELECT current_schema())
```

Comment:
According to the [documentation](https://www.postgresql.org/docs/12/functions-info.html), `current_schema` is meant to be used as a constant, just like `current_user`:

> *Note: current_catalog, current_role, current_schema, current_user, session_user, and user have special syntactic status in SQL: they must be called without trailing parentheses. (In PostgreSQL, parentheses can optionally be used with current_schema, but not with the others.)*


Also, why the sub-select?

```suggestion
           AND n.nspname = current_schema
```

Reply:
Why indeed :smile: :facepalm: ... I made the change. Thanks for the review @odony.